tests/misc/test_miscwidgets.py

```diff
@@ -0,0 +1,97 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2015 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test widgets in miscwidgets module."""
+from unittest import mock
+from PyQt5.QtCore import Qt
+from PyQt5.QtWidgets import QApplication
+import pytest
+
+from qutebrowser.misc import lineparser
+from qutebrowser.misc.miscwidgets import CommandLineEdit
+from qutebrowser.utils import objreg
+
+
+class TestCommandLineEdit:
+    """Tests for CommandLineEdit widget."""
+
+    @pytest.yield_fixture
+    def cmd_edit(self, qtbot):
+        """
+        Fixture to initialize a CommandLineEdit and its dependencies,
+        cleaning up afterwards.
+        """
+        command_history = lineparser.LimitLineParser('', '', limit=None)
+        objreg.register('command-history', command_history)
```

Comment:
I don't think `command-history` is needed here - it's only used for `qutebrowser/mainwindow/statusbar/command.py:Command`, but not in `qutebrowser/misc/miscwidgets.py:CommandLineEdit` which is what we're testing here.